### ISSUE TITLE
Models take too long to compile, use CmdStan's compilation strategy

### ISSUE THREAD

**Issue by riddell-stan** [stan-dev/httpstan#461]
Compiling a model using PyStan takes 5x longer than CmdStan.

**Comment by riddell-stan** [comment]
@syclik , I've got a question about the key bit of code which makes this work:

```
stan::model::model_base& new_model(
        stan::io::var_context& data_context,
        unsigned int seed,
        std::ostream* msg_stream) {
  stan_model* m = new stan_model(data_context, seed, msg_stream);
  return *m;
}
```

Two questions:
- Is this a reference variable to a pointer? Am I reading this right?
- Who is responsible for `free`ing `m`?

**Comment by riddell-stan** [comment]
Addressed by #467
